diff -Nrup kexec-tools-2.0.0.orig/kexec/arch/i386/kexec-x86.c kexec-tools-2.0.0/kexec/arch/i386/kexec-x86.c
--- kexec-tools-2.0.0.orig/kexec/arch/i386/kexec-x86.c	2008-07-14 19:28:19.000000000 +0800
+++ kexec-tools-2.0.0/kexec/arch/i386/kexec-x86.c	2011-04-27 15:31:31.000000000 +0800
@@ -184,6 +184,10 @@ void arch_update_purgatory(struct kexec_
 		&arch_options.console_vga, sizeof(arch_options.console_vga));
 	elf_rel_set_symbol(&info->rhdr, "console_serial",
 		&arch_options.console_serial, sizeof(arch_options.console_serial));
+	elf_rel_set_symbol(&info->rhdr, "backup_src_start",
+		&info->backup_src_start, sizeof(info->backup_src_start));
+	elf_rel_set_symbol(&info->rhdr, "backup_src_size",
+		&info->backup_src_size, sizeof(info->backup_src_size));
 	if (info->kexec_flags & KEXEC_ON_CRASH) {
 		panic_kernel = 1;
 		elf_rel_set_symbol(&info->rhdr, "backup_start",
